lively.morphic/html-morph.js

```diff
@@ -271,3 +271,196 @@ export class IFrameMorph extends HTMLMorph {
     return '<iframe width="100%" height="100%" frameBorder="false" srcdoc=""></iframe>';
   }
 }
+
+/**
+ * A VideoMorph, which is just an overlay on the HTML Video element
+ * This is a subclass of the HTMLMorph, and so it inherits all of the HTMLMorph's
+ * properties.
+ * properties (should be in the constructor)
+ *   - src: url for the video
+ *   - type: type of the video (mp4, ogg, etc)
+ *   - controls: true/false to show not/show the controls (default false)
+ *   - loop: true/false to play on a loop (default true)
+ *   - autoplay: true/false to play on load (default true)
+ *   - autosize:  if true, resize to video size  (default false)
+ *   - autoaspect: if true, resize to correct aspect ratio within current bounding box (default true)
+ * methods:
+ *    - startPlaying() -- start playing the video
+ *    - stopPlaying() -- stop playing the video
+ *    - rewind()  -- set the video counter to 0
+ *
+ * read-only properties:
+ *     - videoDomElement -- the DOM element for the video
+ */
+
+// new VideoMorph().openInWorld()
+export class VideoMorph extends HTMLMorph {
+  static get properties () {
+    return {
+      autoplay: {
+        defaultValue: false,
+        set (aBool) {
+          this.setVideoProperty('autoplay', !!aBool);
+        }
+      },
+      loop: {
+        defaultValue: false,
+        set (aBool) {
+          this.setVideoProperty('loop', !!aBool);
+        }
+      },
+      controls: {
+        defaultValue: false,
+        set (aBool) {
+          this.setVideoProperty('controls', !!aBool);
+        }
+      },
+      src: {
+        defaultValue: false,
+        set (srcURL) {
+          this.setVideoProperty('src', srcURL);
+        }
+      },
+      type: {
+        defaultValue: false,
+        set (videoType) {
+          this.setVideoProperty('type', videoType);
+        }
+      },
+      autosize: {
+        defaultValue: false,
+        set (aBool) {
+          this.setProperty('autosize', !!aBool);
+          this.resizeToExtentProperties();
+        }
+      },
+      autoaspect: {
+        defaultValue: true,
+        set (aBool) {
+          this.setProperty('autosize', !!aBool);
+          this.resizeToExtentProperties();
+        }
+      }
+
+    };
+  }
+
+  constructor (props) {
+    this.videoId = `video-${Math.floor(Math.random() * 1E6)}`;
+    this.constructorProps = props;
+    super(props);
+  }
+
+  setVideoProperty (propertyName, value) {
+    this.setProperty(propertyName, value);
+    this.resetHTML();
+  }
+
+  onLoad () {
+    const props = this.constructorProps;
+    const setField = (field, defaultVal) => {
+      this[field] = props && props[field] ? props[field] : defaultVal;
+    };
+    const normalizeBoolean = field => {
+      this[field] = props ? !!props[field] : false;
+    };
```

Comment:
Well, the values really aren't problematic, I guess, because JavaScript treats any value other than `0`, `undefined`, `null`, and `false` as `false`.  I don't really know how having a non-Boolean value in a Boolean property could hurt, but for display in inspectors, etc., I usually try to normalize them.

Reply:
Got it, makes sense!